### ISSUE TITLE
branching and API experimentation

### ISSUE THREAD

**Issue by odeke-em** [census-instrumentation/opencensus-go#371]
In the current phase where we are getting projects like Go cloud to adopt OpenCensus thus any breaking change here affects them but also requires our APIs to be stable, perhaps let's setup a dev/unstable branch on which we can experiment, prototype, iterate as much as we want and flesh out APIs.

I mention this because I just say a new refactor and package change in https://github.com/census-instrumentation/opencensus-go/pull/370 which we'll need to come to consensus to, but essentially if we didn't have to merge to master, we could prototype with those API changes and instrument some backends, iterate, figure out if there are any kinks and when ready then merge to master.

**Comment by rakyll** [comment]
I'd suggest us to focus on the significant API problems and land improvements to master before this project is becoming more of a dependency for other various projects. Having experimental branches and open PRs are encouraging us to keep these experiments around but not merging things to master.

**Comment by ramonza** [comment]
I agree with @rakyll - we need to push towards API stability. The current set of do-not-merge PRs need agreement before being merged. I don't think it will help to have a single unstable branch that contains multiple changes we aren't yet sure about.

**Comment by ramonza** [comment]
Let's close this for now and see and reconsider in a few weeks.
